k8s-openapi-codegen-common/src/lib.rs

```diff
@@ -999,6 +1006,38 @@ pub fn run(
 		},
 	}
 
+	#[cfg(all(feature = "serde", feature = "schema"))]
+	match &definition.kind {
+		swagger20::SchemaKind::Properties(_) |
+		swagger20::SchemaKind::Ty(
+			swagger20::Type::Object { .. } |
+			swagger20::Type::String { .. } |
+			swagger20::Type::Any |
+			swagger20::Type::IntOrString |
+			swagger20::Type::JsonSchemaPropsOrArray(_) |
+			swagger20::Type::JsonSchemaPropsOrBool(_) |
+			swagger20::Type::JsonSchemaPropsOrStringArray(_)
+		) => {
+			// Inline schema and override the description if present.
+			let re = regex::Regex::new(r##"\{\n\s*"\$ref": "#/definitions/io\.k8s\.([^"]+)"(?:,\n\s*"description": "(.+)")?\n\s*\}"##).unwrap();
```

Comment:
This should use `get_fully_qualified_type_name`, which also handles non-`io.k8s.` paths correctly.